File: js/test_money.js
const assert = require("assert");

class Money {
  constructor(amount, currency) {
    this.amount = amount;
    this.currency = currency;
  }
  times(multiplier) {
    return new Money(this.amount * multiplier, this.currency);
  }

  divide(divisor) {
    return new Money(this.amount / divisor, this.currency);
  }
}

let fiveDollars = new Money(5, "USD");
let tenDollars = new Money(10, "USD");
assert.deepStrictEqual(fiveDollars.times(2), tenDollars);

let originalMoney = new Money(4002, "KRW");
let actualResult = originalMoney.divide(4);
let expectResult = new Money(1000.5, "KRW");
assert.deepStrictEqual(actualResult, expectResult);

File: py/test_money.py
import unittest
from money import Money
from portfolio import Portfolio

class TestMoney(unittest.TestCase):
  def testMultiplication(self):
    tenEuros = Money(10, "EUR")
    twentyEuros = Money(20, "EUR")
    self.assertEqual(twentyEuros, tenEuros.times(2))

  def testDivision(self):
    originalMoney = Money(4002, "KRW")
    expectedMoneyAfterDivision = Money(1000.5, "KRW")
    self.assertEqual(expectedMoneyAfterDivision, originalMoney.divide(4))

  def testAddition(self):
    fiveDollars = Money(5, "USD")
    tenDollars = Money(10, "USD")
    fifteenDollars = Money(15, "USD")
    portfolio = Portfolio()
    portfolio.add(fiveDollars, tenDollars)
    self.assertEqual(fifteenDollars, portfolio.evaluate("USD"))

  def testAdditionOfDollarsAndEuros(self):
    fiveDollars = Money(5, "USD")
    tenEuros = Money(10, "EUR")
    portfolio = Portfolio()
    portfolio.add(fiveDollars, tenEuros)
    exceptedValue = Money(17,"USD")
    actualValue = portfolio.evaluate("USD")
    self.assertEqual(
      exceptedValue,actualValue, "%s != %s" % (exceptedValue, actualValue)
    )


if __name__ == '__main__':
    unittest.main()